lib/vagrant/ui.rb

```diff
@@ -75,6 +80,12 @@ def color?
       def machine(type, *data)
         @logger.info("Machine: #{type} #{data.inspect}")
       end
+
+      def output_if_showing_progress
```

Comment:
Maybe we can rename this to `rewriting`? Then, when used within the code we'd end up with something like:

```ruby
loop do
  @ui.rewriting do |ui|
    ui.detail "Status ..."
    ui.clear_line
  end
  sleep 5
end
```

The `rewriting` method would then simply `yield` self, except in the new non interactive subclass, which would provide a no-op.